### ISSUE TITLE
Another "macro not expanded" error with implicits, inferred Nothing, default args

### ISSUE THREAD

**Issue by scabug** [scala/bug#10073]
```scala
class Yo[Unused] {
  def yo(hasDefault: Any = ""): String = ""
}

class MacroNotExpanded {
  implicit def toYo[Unused](a: Any)(implicit ct: reflect.ClassTag[Unused]): Yo[Unused] = new Yo[Unused]
   "".yo()  
}
```

```
/Users/jz/code/scratch/src/main/scala/Main.scala:7: error: macro has not been expanded
   "".yo()
   ^
one error found
```

Same symptom and similar causes to #7987

**Comment by retronym** [comment]
I've taken another look at this problem.

I first tried to change the use of the global map `delayed: Map[Tree, _]` to use of a tree attachment, to have a better chance of surviving tree duplication. But I'm not actually sure if that is a factor in the test cases we have here.

I then found that in `macroExpandAll` (the code the looks at all subtrees of each typechecked tree for any delayed expansions that are ready to be expanded) to properly issue errors that the macro issues.

https://github.com/scala/scala/compare/2.12.x...retronym:ticket/10073?expand=1
